src/DefaultEditor.js

```diff
@@ -17,8 +17,58 @@ import {
   StyleUpdateMenusPanel,
 } from './default_panels';
 import Logo from './components/widgets/Logo';
+import {TRANSFORMABLE_TRACES} from './lib/constants';
 
 class DefaultEditor extends Component {
+  constructor(props, context) {
+    super(props, context);
+    this.hasTransforms = this.hasTransforms.bind(this);
+    this.hasAxes = this.hasAxes.bind(this);
+    this.hasMenus = this.hasMenus.bind(this);
+    this.hasSliders = this.hasSliders.bind(this);
+    this.hasColorbars = this.hasColorbars.bind(this);
+  }
+
+  hasTransforms() {
+    return this.context.fullData.some(d =>
+      TRANSFORMABLE_TRACES.includes(d.type)
+    );
+  }
+
+  hasAxes() {
+    return (
+      Object.keys(this.context.fullLayout._subplots).filter(
+        type =>
+          !['cartesian', 'mapbox'].includes(type) &&
+          this.context.fullLayout._subplots[type].length > 0
+      ).length > 0
+    );
+  }
+
+  hasMenus() {
+    const {
+      fullLayout: {updatemenus = []},
+    } = this.context;
+
+    return updatemenus.length > 0;
+  }
+
+  hasSliders() {
+    const {
+      layout: {sliders = []},
+    } = this.context;
+
+    return sliders.length > 0;
+  }
+
+  hasColorbars() {
+    return this.context.fullData.some(
+      d =>
+        (d.marker && d.marker.showscale !== undefined) || // eslint-disable-line no-undefined
+        d.showscale !== undefined // eslint-disable-line no-undefined
```

Comment:
This is good. I know the panel still appears even when all traces are in constant-color mode, but that's ok. at least this panel never appears when there's nothing you can do in it :)

Reply:
once you saw that panel appear, you don't wanna loose it again right? :)